clang/docs/SYCLFERFCIntegrationHeaderFooter.md

```diff
@@ -0,0 +1,179 @@
+# Integration Header
+
+We propose an Integration Header for two purposes:
+
+* to allow a non-SYCL-aware compiler to do host compilations and
+* as a mechanism permitting the SYCL run-time library to use the information provided for its needs
```

Comment:
I think some additional introductory material would be helpful. How does the following sound?

```suggestion
# Introduction

SYCL source files are logically (and usually physically) compiled multiple times; code that runs on the "host" (e.g., a CPU) is compiled by a host compiler and code that runs on a "device" (e.g., a GPU) is compiled by a device compiler. The SYCL language is designed such that it is not necessary that the host compiler be SYCL-aware; an ISO C++ conforming compiler suffices. However, some SYCL features such as kernel invocation and support for
[specialization constants](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_specialization_constants)
require coordination between the host and device compilers. Integration headers and footers produced by a device compiler are used to communicate the information needed during host compilation to support these features.

# Integration Header

An integration header is a source file produced during device compilation that is intended to be pre-included during host compilation of the same source file.
```

Reply:
Thanks, done.